File: csharp/Structural/06-Adapter/Adapter/SmsNotifierAdapter.cs
// Object Adapter (composition) — recommended approach.
// Wraps LegacySmsGateway and implements INotifier so the client
// never needs to know the legacy class exists.

public class SmsNotifierAdapter : INotifier
{
    private readonly LegacySmsGateway _gateway;
    private readonly string           _phoneNumber;

    public SmsNotifierAdapter(LegacySmsGateway gateway, string phoneNumber)
    {
        _gateway     = gateway;
        _phoneNumber = phoneNumber;
    }

    public void SendNotification(string jobId, string message)
    {
        var smsText = $"[FixIt Pro] Job #{jobId}: {message}";
        _gateway.Transmit(_phoneNumber, smsText);
    }
}

File: typescript/Structural/06-Adapter/Adapter/SmsNotifierAdapter.ts
import { INotifier } from "../Notifier/INotifier";
import { LegacySmsGateway } from "../Legacy/LegacySmsGateway";

// Object Adapter — wraps LegacySmsGateway, implements INotifier.

export class SmsNotifierAdapter implements INotifier {
  constructor(
    private readonly gateway: LegacySmsGateway,
    private readonly phoneNumber: string,
  ) {}

  sendNotification(jobId: string, message: string): void {
    const smsText = `[FixIt Pro] Job #${jobId}: ${message}`;
    this.gateway.transmit(this.phoneNumber, smsText);
  }
}
